include/seqan3/argument_parser/validators.hpp

```diff
@@ -309,68 +328,260 @@ class file_ext_validator
     //!\endcond
     void operator()(range_type const & v) const
     {
-        std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
+         std::for_each(v.begin(), v.end(), [&] (auto cmp) { this->operator()(cmp); });
     }
 
+private:
+
+    //!\brief Stores the extensions.
+    std::vector<std::string> extensions{};
+};
+
+/*!\brief A validator that checks if a given path is a valid input file.
+ * \ingroup argument_parser
+ * \implements seqan3::validator_concept
+ *
+ * \details
+ *
+ * On construction, the validator must receive a list (std::vector over std::string) of valid file extensions.
```

Comment:
```suggestion
 * On construction, the validator may receive a list (std::vector over std::string) of valid file extensions.
```

You document that it **must** receive a list, but you provide a default constructor.

Reply:
Copy'n'paste issue